### DISCUSSION TITLE
Support for configuring SMTP email at project level

### DISCUSSION THREAD

**Discussion by rvndbalaji** [langfuse/langfuse#3056] [Support] [answered]
### Describe the feature or potential improvement

Support for configuring SMTP email within the application

1. As an admin I would like to configure one SMTP server per project 
2. I would also require the ability to configure SMTP port number and smtp_starttls
3. I would like the configuring of email within the application rather than at the env variable

**Comment by marcklingen** [comment]
This is unfortunately outside of the scope of the project as these credentials are to be maintained on an instance level and should not be exposed to users or project admins. Is it an option for you to set this at the instance level?

**Reply by rvndbalaji** [reply]
@marcklingen 
I did set it at the instance level, but my smpt has the following configuration 

   ```
 sender = "no-reply@company.com"
    smtp_server = "smartmail.company.net"
    smtp_port = 25
    smtp_starttls = True
```

When I do  put this in the config, there is no option to set the PORT number
I get the following error 
```

Error: connect ECONNREFUSED 127.0.0.1:587     at TCPConnectWrap.afterConnect [as oncomplete] (node:net:1607:16)     at TCPConnectWrap.callbackTrampoline (node:internal/async_hooks:130:17) {
  errno: -111,
  code: 'ESOCKET',
  syscall: 'connect',
  address: '127.0.0.1',
  port: 587,
  command: 'CONN'
```

Im not sure why this is happening, when I call the same config via python to send an email it works perfectly from the same pod
Any ideas?

**Reply by marcklingen** [reply, marked as answer]
how did you turn this into a connection string?

we pass the connection string to `parseConnectionUrl` of `nodemailer`, more on this here: https://nodemailer.com/smtp/#:~:text=Alternatively%2C%20you%20could%20use%20a%20connection%20url%20instead%20of%20an%20object%20for%20the%20options.%20Use%20%E2%80%98smtp%3A%27%20or%20%E2%80%98smtps%3A%27%20as%20the%20protocol%20in%20the%20url.

**Reply by rvndbalaji** [reply]
OHHHHHH! 

In the docs it says 
SMTP_CONNECTION_URL |   | Configure optional SMTP server connection for transactional email.

I totally missed the URL part, I put the SMTP host instead !! This was stupid of me
I tried it with this -`smtp://smartmail.company.net:25?starttls=true` and it worked

(Maybe we could add an example in the docs?)

Thank you

**Reply by marcklingen** [reply]
Awesome, glad that it works! I updated the docs to include a reference to nodemailer: https://github.com/langfuse/langfuse-docs/commit/607f29df43e5e3e0f23e13ec979f580204b6ea72

**Reply by rvndbalaji** [reply]
You guys are quick. I just started using LangFuse and I have some suggestions as I came over from LangSmith, 
Ill start a new thread

**Reply by marcklingen** [reply]
Thanks! Happy to help

Please add new ideas here: langfuse.com/ideas
